hbase-server/src/main/java/org/apache/hadoop/hbase/wal/AbstractWALRoller.java

```diff
@@ -115,9 +113,9 @@ protected AbstractWALRoller(String name, Configuration conf, T abortable) {
    */
   private void checkLowReplication(long now) {
     try {
-      for (Entry<WAL, Boolean> entry : walNeedsRoll.entrySet()) {
+      for (Entry<WAL, RollController> entry : wals.entrySet()) {
         WAL wal = entry.getKey();
-        boolean needRollAlready = entry.getValue();
+        boolean needRollAlready = entry.getValue().isRollRequested();
```

Comment:
Previously, when we are going for a roll because of periodic roll, then also we used to keep the value in Map as TRUE right?  Means here we checked that also.  Because of any reason, the WAL is marked for roll, dont do this check.  So here that is missing now.